src/main.cpp

```diff
@@ -4885,19 +4885,14 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     }
 
 
-    if (!(nLocalServices & NODE_BLOOM) &&
+    if (!pfrom->fWhiteListed && !(nLocalServices & NODE_BLOOM) &&
               (strCommand == NetMsgType::FILTERLOAD ||
                strCommand == NetMsgType::FILTERADD ||
-               strCommand == NetMsgType::FILTERCLEAR))
-    {
-        if (pfrom->nVersion >= NO_BLOOM_VERSION) {
-            LOCK(cs_main);
-            Misbehaving(pfrom->GetId(), 100);
-            return false;
-        } else {
-            pfrom->fDisconnect = true;
-            return false;
-        }
+               strCommand == NetMsgType::FILTERCLEAR)) {
+        LOCK(cs_main);
+        Misbehaving(pfrom->GetId(), 5);
```

Comment:
The idea here is that it will be banned if it continues to connect and request bloom services 20 times... I suspect the misbehave count gets reset upon disconnection though..!